sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/dag/ErroredDependencyTaskException.java

```diff
@@ -6,5 +6,5 @@
 /**
  * An FaultedDependencyException is emitted when a task cannot be executed due to a faulted decedent task.
  */
-final class ErroredDependencyTaskException extends Exception {
+final class ErroredDependencyTaskException extends RuntimeException {
```

Comment:
It's not expected for user to catch this Exception. It'll be caught only in `TaskGroup` implementation.